mountpoint-s3-client/tests/put_object.rs

```diff
@@ -375,6 +376,66 @@ async fn test_put_checksums(trailing_checksums: PutObjectTrailingChecksums) {
     }
 }
 
+#[test_case(HashMap::new(); "Empty")]
+#[test_case(HashMap::from([("foo".to_string(), "bar".to_string()), ("a".to_string(), "b".to_string())]); "ASCII")]
+#[tokio::test]
+async fn test_put_user_object_metadata_happy(object_metadata: HashMap<String, String>) {
+    const PART_SIZE: usize = 5 * 1024 * 1024;
```

Comment:
(Here and below) Why not upload a small/empty object? Even just configuring `PART_SIZE` is just noise for the actual test.